tuning/acceleration/plugins/framework_plugin_unsloth_autogptq.py

```diff
@@ -0,0 +1,150 @@
+from auto_gptq.utils.peft_utils import GPTQLoraModel
+from unsloth import FastLanguageModel
+from unsloth.utils.modeling import QuantizationMethod
+from unsloth.gptq.triton.layers import GPTQuantLinear
+
+from types import MethodType
+from typing import Tuple, List, Dict
+
+from .framework_plugin import AccelerationPlugin
+from .framework_plugin_autogptq import _replace_module, LoraLinearGPTQ
+
+# def _dispatch_lora_linear_for_gptq(
+def _create_new_module_triton(
+    lora_config: LoraConfig,
+    adapter_name: str,
+    target: torch.nn.Module,
+    **kwargs,
+):
+    # if the base layer module matches a supported class, dispatch the lora linear
+    # to be installed
+    new_module = target
+    if (
+        isinstance(target, GPTQuantLinear)
+    ):
+        new_module = LoraLinearGPTQ(target, adapter_name, **kwargs)
+    # if module cannot be found, return None which results in a raise in the call-stack
+    return new_module
+
+class UnslothAutoGPTQAccelerationPlugin(AccelerationPlugin):
+    
+    require_packages = ['auto_gptq', 'unsloth']
+
+    def __init__(self, configurations: Dict[str, Dict]):
+        super().__init__(configurations)
+
+        # just do checking, nothing must to configure at this point
+        # if need to configure then do something like this:
+        # self.kernel = self._get_config_value("peft.kernel")
+        self._check_config_equal(key="peft.quantization", value="unsloth_autogptq")
+        self._check_config_equal(key="peft.quantization.unsloth_autogptq.kernel", value="triton_v2")
+
+    def model_loader(self, model_name: str, **kwargs):
+        # 1. Load the gptq base model through unsloth FastLanguageModel
+        model, _ = FastLanguageModel.from_pretrained(
+            model_name,
+            quantization_method=QuantizationMethod.GPTQ,
+            device_map="auto",
+        )
+
+        # 2. Depending on the model, replace appropriate CUDA kernel with TritonV2 kernel 
+        if model.config.model_type == 'mixtral':
+            from auto_gptq.nn_modules.qlinear.qlinear_cuda_old import (
+                QuantLinear as QuantLinearCuda,
+            )                
+        else:
+            from auto_gptq.nn_modules.qlinear.qlinear_cuda import (
+                QuantLinear as QuantLinearCuda,
+            )
+        GPTQuantLinear.inject_to_model(
+            model, target_module_type=QuantLinearCuda
+        )
+
+        return model
+
+    @property
+    def requires_custom_loading(self):
+        return True
+
+    @property
+    def requires_agumentation(self):
+        return True
+
+    def augmentation(
+        self, 
+        model, 
+        train_args: TrainingArguments,
+        modifiable_args: Tuple[LoraConfig],
+    ):
+        peft_config, = modifiable_args # unpack modifiable args
+
+        # some assertions
+        assert peft_config is not None, "need peft_config to install PEFT adapters"
+        assert model.dtype == torch.float16 or train_args.fp16,\
```

Comment:
assert the lora_dropout here, just like we did when checking `peft_config is None`